book/src/qobject/rustobj_struct.md

```diff
@@ -20,12 +20,12 @@ The RustObj struct allows you to define the following items
 
 ## Invokables
 
-To define a method which is exposed to QML and C++, add a method on the `RustObj` struct and add the attribute `#[invokable]`. The parameters and return type are then matched to the Qt side. Also CXX-Qt automatically adds wrapper code around your invokable to automatically perform any conversion between the [C++ and Rust types](../concepts/types.md).
+To define a method which is exposed to QML and C++, add a method on the `RustObj` struct, with `impl Qt<RustObj>`, and add the attribute `#[invokable]`. The parameters and return type are then matched to the Qt side. Also CXX-Qt adds wrapper code around your invokable to automatically perform any conversion between the [C++ and Rust types](../concepts/types.md).
 
 Note to access properties on the C++ object use [Cpp Object](./cpp_object.md).
 
 ## Private Methods and Fields
 
 Unlike the [Data Struct](./data_struct.md) fields which are defined on the `RustObj` struct are not exposed as properties to Qt. These can be considered as "private to Rust" fields, and are useful for storing channels for threading or internal information for the QObject.
 
-Methods implemented on the `RustObj` that do not have an `#[invokable]` attribute are not exposed to C++ and are considered "private to Rust" methods. Similar to fields these are useful for threading and internal information.
+Methods implemented on the `RustObj` (and not the `impl Qt<RustObj>`) that do not have an `#[invokable]` attribute are not exposed to C++ and are considered "private to Rust" methods. Similar to fields these are useful for threading and internal information.
```

Comment:
Maybe rephrase this a bit to focus on the access to C++ features.
> Methods implemented using `impl RustObj` (and not `impl Qt<RustObj>`) are just normal Rust member methods.
> Therefore they do not have access to any C++/QObject functionality (e.g. emitting Signals, changing properties, etc.)
> You will usually only need to use `impl RustObj` if you want to also use your `RustObj` struct as a normal Rust struct, that is not wrapped in a QObject.

Reply:
Done